#!/usr/bin/env python
PACKAGE = "grasp_utilities"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("number_fingers",    int_t,    0, "Numero de dedos que se usan", 5,  2, 5)
gen.add("min_pressure_threshold", double_t, 0, "Umbral minimo de presion", 2.0, 0.5, 3.0)
gen.add("max_pressure_threshold", double_t, 0, "Umbral maximo de presion", 3.0, 2.0, 7.0)
gen.add("position_step",double_t, 0, "Paso a desplazar articulaciones en el proceso de reajuste", 0.025, 0.0005, 0.050)
gen.add("tip_position_step", double_t, 0, "Paso a desplazar articulacion del extremo del dedo", 0.025, 0.0005, 0.050)
gen.add("thumb_step", double_t, 0, "Paso a desplazar el pulgar", 0.0, 0.0, 0.05)
gen.add("wrist_step", double_t, 0, "Paso a desplazar articulacion de la munheca", -0.002, -0.01, 0.0)
gen.add("little_step", double_t, 0, "Paso a desplazar el dedo pequenho", 0.0, 0.0, 0.78)
gen.add("wrist_angle_x",double_t, 0, "Angulo de la muneca al iniciar el grasp", 0.21, -0.77, 0.50)
gen.add("wrist_angle_y",double_t, 0, "Angulo de la muneca al iniciar el grasp", 0.0, -0.50, 0.17)
gen.add("knuckle_aperture", double_t, 0, "Distancia inicial hueco mano", 0.0, 0.0,  1.57)
gen.add("middle_aperture", double_t, 0, "Posicion inicial tips", 0.575, 0.0,  1.57)
gen.add("hub_aperture", double_t, 0, "Distancia entre dedos", 0.0, -0.349, 0.0)
gen.add("little_finger_position", double_t, 0, "Distancia entre dedos", 0.0, 0.0, 0.70)
gen.add("write_files", int_t, 0, "Activar / desactivar escritura en archivo", 0, 0, 1)

gen.add("pa10_shadow_goal_x", double_t, 0, "Goal pose (x)", 0.0, -2.0,  2.0)
gen.add("pa10_shadow_goal_y", double_t, 0, "Goal pose (y)", 0.0, -2.0,  2.0)
gen.add("pa10_shadow_goal_z", double_t, 0, "Goal pose (z)", 0.0, -2.0,  2.0)
gen.add("pa10_shadow_goal_w", double_t, 0, "Goal pose (w)", 0.0, -2.0,  2.0)

exit(gen.generate(PACKAGE, "grasp_utilities", "grasp_reconfiguration"))